internal/appconfig/validation.go

```diff
@@ -320,3 +321,39 @@ func (cfg *Config) validateMounts() (extraInfo string, err error) {
 	}
 	return
 }
+
+func (cfg *Config) validateRestartPolicy() (extraInfo string, err error) {
+	if cfg.Restart == nil {
+		return
+	}
+
+	for _, restart := range cfg.Restart {
+		validGroupNames := cfg.ProcessNames()
+
+		// first make sure restart.Processes matches a valid process name.
+		for _, processName := range restart.Processes {
+			if !slices.Contains(validGroupNames, processName) {
+				extraInfo += fmt.Sprintf("Restart policy specifies '%s' as one of its processes, but no processes are defined with that name; "+
+					"update fly.toml [processes] to add '%s' process or remove it from restart policy's processes list\n",
+					processName, processName,
+				)
+				err = ValidationError
+			}
+		}
+
+		switch restart.Policy {
```

Comment:
you can reuse `parseRestartPolicy` function here to keep validation and parsing in sync.